glambie/data/submission_cleaning_script.py

```diff
@@ -0,0 +1,482 @@
+
+This script should be run after all submissions have been received for a single round of the GlaMBIE project, before the
+algorithm is run.
+"""
+
+from datetime import datetime
+import logging
+import os
+
+from google.cloud.storage import Client
+import numpy as np
+import pandas as pd
+from pathlib import Path
+from typing import Tuple
+
+from glambie.monitoring.logging import setup_logging
+from glambie.util.date_helpers import datetime_dates_to_fractional_years
+from glambie.util.timeseries_helpers import interpolate_change_per_day_to_fill_gaps
+
+DATA_TRANSFER_BUCKET_NAME = "glambie-submissions"
+MAX_ALLOWED_ELEVATION_CHANGE_M = 100
+MAX_ALLOWED_ELEVATION_CHANGE_GT = 10000
+log = logging.getLogger(__name__)
+
+
+def download_csv_files_from_bucket(storage_client: Client, local_data_directory_path: str,
+                                   region_prefix: str = None) -> list[str]:
+    """
+    Function to download glambie .csv files from the google bucket to a local folder, where they can be checked and
+    edited. Specify files for a specific region using the region_prefix parameter - otherwise all .csv files in the
+    bucket will be downloaded.
+
+    Parameters
+    ----------
+    storage_client : Client
+        Google Cloud storage Client instance that will be used to access bucket (? not sure what this actually does)
+    local_data_directory_path : str
+        Path to save local copies of bucket data to.
+    region_prefix : str, Optional
+        String describing the region name pattern to look for in the filenames when deciding which to download,
+        by default None.
+
+    Returns
+    -------
+    list[str]
+        List of files that have been downloaded to the local directory.
+    """
+    list_of_blobs_in_bucket = storage_client.list_blobs(DATA_TRANSFER_BUCKET_NAME, prefix=region_prefix)
+    downloaded_files = []
+
+    for blob in list_of_blobs_in_bucket:
+        if '.csv' in blob.name:
+            downloaded_files.append(blob.name)
+            destination_file_path = os.path.join(local_data_directory_path, blob.name)
+            # check that destination directory exists and create it if it doesn't
+            Path(destination_file_path).parent.mkdir(parents=True, exist_ok=True)
+            with open(destination_file_path, "wb") as output_file:
+                blob.download_to_file(output_file, raw_download=False)
+
+    return downloaded_files
+
+
+def upload_edited_csv_files_to_bucket(storage_client: Client, files_to_upload: list[str], local_path: str):
+    """
+    After editing local copies of the submitted csv files, replace the original versions in the bucket with the
+    edited local versions. Also upload an archive folder containing original copies of all edited files
+
+    Parameters
+    ----------
+    storage_client : Client
+        Google Cloud storage Client instance that will be used to access bucket (? not sure what this actually does)
```

Comment:
```suggestion
        Google Cloud storage Client instance that will be used to access bucket.
```

Reply:
The client object encapsulates all of the logic to talk to google buckets. Handles things like security and other opaque API stuff that we as python developers don't need to care about.